src/com/owncloud/android/widgets/ShortcutsWidgetConfigureActivity.java

```diff
@@ -0,0 +1,146 @@
+        setResult(RESULT_CANCELED);
+
+          // Find the widget id from the intent.
+        Intent intent = getIntent();
+        Bundle extras = intent.getExtras();
+        if (extras != null) {
+            mAppWidgetId = extras.getInt(
+                    AppWidgetManager.EXTRA_APPWIDGET_ID, AppWidgetManager.INVALID_APPWIDGET_ID);
+
+        }
+
+        // If this activity was started with an intent without an app widget ID,
+        // finish with an error.
+        if (mAppWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID) {
+            finish();
+            return;
+        }
+
+        // Show dialog with the accounts
+        AccountsDialogFragment accountsDialogFragment = new AccountsDialogFragment().getInstance();
+        accountsDialogFragment.setOnItemClickListener(this);
+        Fragment fr = getFragmentManager().findFragmentByTag(AccountsDialogFragment.TAG);
+        if (fr == null) {
+            accountsDialogFragment.show(getFragmentManager(), AccountsDialogFragment.TAG);
+        }
+
+    }
+
+
+    // Write the prefix to the SharedPreferences object for this widget
+    static void saveAccountPref(Context context, int appWidgetId, String text) {
+        SharedPreferences.Editor prefs = context.getSharedPreferences(PREFS_NAME, 0).edit();
+        prefs.putString(PREF_PREFIX_KEY + appWidgetId, text);
+        prefs.commit();
+    }
+
+    // Read the prefix from the SharedPreferences object for this widget.
+    // If there is no preference saved, get the default from a resource
+    static String loadAccountPref(Context context, int appWidgetId) {
+        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, 0);
+        String userValue = prefs.getString(PREF_PREFIX_KEY + appWidgetId, null);
+        if (userValue != null) {
+            return userValue;
+        } else {
+            return context.getString(R.string.username);
+        }
+    }
+
+    static void deleteAccountPref(Context context, int appWidgetId) {
+        SharedPreferences.Editor prefs = context.getSharedPreferences(PREFS_NAME, 0).edit();
+        prefs.remove(PREF_PREFIX_KEY + appWidgetId);
+        prefs.commit();
+    }
+
+    private void configureResult(){
+
+        Context context = ShortcutsWidgetConfigureActivity.this;
+
+        // It is the responsibility of the configuration activity to update the app widget
+        AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(context);
+        ShortcutsWidget.updateAppWidget(context, appWidgetManager, mAppWidgetId);
+
+        // Make sure we pass back the original appWidgetId
+        Intent resultValue = new Intent();
+        resultValue.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, mAppWidgetId);
+        setResult(RESULT_OK, resultValue);
+        finish();
+    }
+
+    @Override
+    public void onAccountSelected(String accountName) {
+        Context context = ShortcutsWidgetConfigureActivity.this;
```

Comment:
this is not inner class ShortcutsWidgetConfigureActivity.this is not needed, and Activity is a Context